### PR TITLE
6006 metadata inconsistencies if owner assignment fails because of table movement

### LINKED ISSUE #6006: Metadata inconsistencies if owner assignment fails because of table movement
When you assign an organization owner his tables are moved from `public` to `username` schema. If one table movement fails process is reverted, keeping the tables in `public`. Nevertheless there're some problems:

- [X] user schema (`users.database_schema`) is changed.
- [X] Username database schema should be deleted.
- [ ] organization should be deleted in the box.

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #6006.

### CODE REVIEW

**Comment by juanignaciosl** [comment]
@Kartones CR this, please (last test failed only because test organization name collision).

File: app/controllers/superadmin/organizations_controller.rb [hunk 1]
```diff
@@ -25,6 +25,15 @@ def create
       uo.promote_user_to_admin
     end
     respond_with(:superadmin, @organization)
+  rescue => e
+    begin
+      @organization.delete if @organization
+    rescue => e
+      # Avoid shadowing original error
+      Rollbar.report_message('Cleaning failed creation', 'error', error: e.inspect, organization: @organization)
```
Comment (Kartones) [inline app/controllers/superadmin/organizations_controller.rb]: Why not using `CartoDB.notify_error` ?

**Comment by Kartones** [comment]
:+1:

File: app/controllers/superadmin/organizations_controller.rb (hunk 1 above)
Comment (juanignaciosl) [inline app/controllers/superadmin/organizations_controller.rb]: Absolutely. I copied another notification in that file and didn't realise it was an old, Rollbar one.

**Pull request merged**